### PR TITLE
[docs] Add alt text on images (related to #4036)

### PR DESCRIPTION
**Pull request by akshitadixit**
This pull request mentions #4028 and modifies the following files only:

- [x] docs/Features.rst
- [x] docs/GPU-Performance.rst
- [x] docs/GPU-Windows.rst
- [ ] docs/Parallel-Learning-Guide.rst
- [ ] docs/Parameters-Tuning.rst
- [ ] docs/gcc-Tips.rst
- [ ] docs/index.rst

### CODE REVIEW

**Comment by jameslamb** [comment]
I apologize for the inconvenience, but could you please merge in the latest changes from `master`? We recently merged some fixes for failing continuous integration jobs (#4032)

**Comment by akshitadixit** [comment]
Sure

**Comment by akshitadixit** [comment]
@jameslamb if you could please let me know if the checks are failing on my end?

**Comment by akshitadixit** [comment]
Definitely, thanks for informing.

Also I came across these articles and then chose full stops instead of simple spaces. Also tried it out with a tts to confirm.
* [article](https://accessibility.blog.gov.uk/2017/02/08/advice-for-creating-content-that-works-well-with-screen-readers/#:~:text=A%20screen%20reader%20might%20speak,.%22%20is%20an%20interesting%20case)
* [article](https://accessibility.psu.edu/abbreviations/)

File: docs/GPU-Windows.rst [hunk 1]
```diff
@@ -397,6 +411,7 @@ Now that you compiled LightGBM, you try it... and you always see a segmentation
 .. image:: ./_static/images/screenshot-segmentation-fault.png
    :align: center
    :target: ./_static/images/screenshot-segmentation-fault.png
+   :alt: A screenshot of the command prompt where a segmentation fault has occured while using Light B G M
```
Comment (jameslamb) [inline docs/GPU-Windows.rst:414]:
```suggestion
   :alt: A screenshot of the command prompt where a segmentation fault has occurred while using Light G B M
```

I make this mistake a lot in this project haha

File: docs/GPU-Windows.rst [hunk 2]
```diff
@@ -366,6 +378,7 @@ You can do everything in the Git Bash console you left open:
 .. image:: ./_static/images/screenshot-lightgbm-with-gpu-support-compiled.png
    :align: center
    :target: ./_static/images/screenshot-lightgbm-with-gpu-support-compiled.png
+      :alt: A screenshot of the gitbash window with Light G B M successfully installed
```
Comment (jameslamb) [inline docs/GPU-Windows.rst:381]:
```suggestion
      :alt: A screenshot of the git bash window with Light G B M successfully installed
```

File: docs/GPU-Windows.rst [hunk 3]
```diff
@@ -270,6 +277,7 @@ Installing CMake requires one download first and then a lot of configuration for
 .. image:: ./_static/images/screenshot-downloading-cmake.png
    :align: center
    :target: ./_static/images/screenshot-downloading-cmake.png
+   :alt: A screenshot of the binary distributions of c-make for downloading on Windows 64 bit
```
Comment (jameslamb) [inline docs/GPU-Windows.rst:280]:
```suggestion
   :alt: A screenshot of the binary distributions of C Make for downloading on 64 bit Windows
```

I think the way you separated this out into `C` and `Make` will be better understood by screen readers.

File: docs/GPU-Performance.rst [hunk 4]
```diff
@@ -163,6 +163,7 @@ We record the wall clock time after 500 iterations, as shown in the figure below
 .. image:: ./_static/images/gpu-performance-comparison.png
    :align: center
    :target: ./_static/images/gpu-performance-comparison.png
+   :alt: A G. P. U. performance chart which is a record of the wall clock time after 500 iterations for Higgs, epsilon, Bosch, Microsoft L. T. R, Expo and Yahoo L. T. R. and bin size of 63 performs comparitively better.
```
Comment (jameslamb) [inline docs/GPU-Performance.rst:166]:
```suggestion
   :alt: A performance chart which is a record of the wall clock time after 500 iterations on G P U for Higgs, epsilon, Bosch, Microsoft L T R, Expo and Yahoo L T R and bin size of 63 performs comparitively better.
```

Based on this article you shared (https://accessibility.psu.edu/abbreviations/, thanks for sharing that!), I think that for these `alt` tags all acronyms should just use spaces and not full stops.

> When writing ALT tags with acronyms, add spaces between characters (e.g. <alt="P S U" >).

I also think that for this particular one, the "GPU" part should be moved to the middle of the sentence so it's clear that "AGPU" is not a meaningful acronym.

File: docs/GPU-Windows.rst [hunk 5]
```diff
@@ -105,6 +108,7 @@ You may choose a version other than the most recent one if you need a previous M
 .. image:: ./_static/images/screenshot-mingw-installation.png
    :align: center
    :target: ./_static/images/screenshot-mingw-installation.png
+   :alt: A screenshot of the Min G. W. installation setup settings window
```
Comment (jameslamb) [inline docs/GPU-Windows.rst:111]:
```suggestion
   :alt: A screenshot of the Min G W installation setup settings window
```

File: docs/GPU-Windows.rst [hunk 6]
```diff
@@ -123,6 +127,7 @@ You can check which MinGW version you are using by running the following in a co
 .. image:: ./_static/images/screenshot-r-mingw-used.png
    :align: center
    :target: ./_static/images/screenshot-r-mingw-used.png
+   :alt: A screenshot of the administrator command prompt where G. C. C. version is being checked
```
Comment (jameslamb) [inline docs/GPU-Windows.rst:130]:
```suggestion
   :alt: A screenshot of the administrator command prompt where G C C version is being checked 
```

**Review by jameslamb — CHANGES_REQUESTED** [review]
Thanks so much for taking the time to do this, and for your patience with me as it took a long time for me to come back and give you a review.

Could you please consider the suggestions I just left? You can click "add to batch" in the browser and then "commit" after adding all of them to directly commit them, if you agree.

Can you also either merge in the latest `master` to this branch? Sorry for the inconvenience. We've had some issues recently with our automated continuous integration tests, and recent pull requests fixed those issues (I hope haha).

**Comment by akshitadixit** [comment]
Sure! Thanks for the suggestions, will do it.

File: docs/GPU-Windows.rst (hunk 1 above)
Comment (akshitadixit) [reply docs/GPU-Windows.rst:414]: Dang T_T

**Comment by jameslamb** [comment]
@akshitadixit it looks like this branch has some conflicts with recent changes on `master`. I'd be happy to fix them if you'd like (they aren't related to your code). Can I push to your branch?

**Comment by akshitadixit** [comment]
> @akshitadixit it looks like this branch has some conflicts with recent changes on `master`. I'd be happy to fix them if you'd like (they aren't related to your code). Can I push to your branch?

Sure, please do!

**Review by jameslamb — APPROVED** [review]
Thanks so much! These changes look great to me.

Are you interested in helping us to finish https://github.com/microsoft/LightGBM/issues/4028? I'd be grateful for another pull request that adds alt text for any images in `docs/Parallel-Learning-Guide.rst` and `docs/gcc-Tips.rst`, if you have time and interest.

**Comment by akshitadixit** [comment]
Oh definitely! Will finish it off.

File: docs/Features.rst [hunk 7]
```diff
@@ -53,6 +54,7 @@ Leaf-wise may cause over-fitting when ``#data`` is small, so LightGBM includes t
 
 .. image:: ./_static/images/leaf-wise.png
    :align: center
+   :alt: A diagram depicting leaf wise tree growth in which only the node with the highest loss change is split and not bother with the rest of the nodes in the same level. This results in an asymmetrical tree where subsequent splitting is happenning only on one side of the tree.
```
Comment (StrikerRUS) [inline docs/Features.rst:57]:
```suggestion
   :alt: A diagram depicting leaf wise tree growth in which only the node with the highest loss change is split and not bother with the rest of the nodes in the same level. This results in an asymmetrical tree where subsequent splitting is happening only on one side of the tree.
```

File: docs/GPU-Performance.rst [hunk 8]
```diff
@@ -163,6 +163,7 @@ We record the wall clock time after 500 iterations, as shown in the figure below
 .. image:: ./_static/images/gpu-performance-comparison.png
    :align: center
    :target: ./_static/images/gpu-performance-comparison.png
+   :alt: A performance chart which is a record of the wall clock time after 500 iterations on G P U for Higgs, epsilon, Bosch, Microsoft L T R, Expo and Yahoo L T R and bin size of 63 performs comparitively better.
```
Comment (StrikerRUS) [inline docs/GPU-Performance.rst:166]:
```suggestion
   :alt: A performance chart which is a record of the wall clock time after 500 iterations on G P U for Higgs, epsilon, Bosch, Microsoft L T R, Expo and Yahoo L T R and bin size of 63 performs comparatively better.
```

File: docs/GPU-Windows.rst [hunk 9]
```diff
@@ -46,18 +46,21 @@ To modify PATH, just follow the pictures after going to the ``Control Panel``:
 .. image:: ./_static/images/screenshot-system.png
    :align: center
    :target: ./_static/images/screenshot-system.png
+   :alt: A screenshot of the System option under System and Security of the Control Panel
```
Comment (StrikerRUS) [inline docs/GPU-Windows.rst:49]:
```suggestion
   :alt: A screenshot of the System option under System and Security of the Control Panel.
```

File: docs/GPU-Windows.rst [hunk 10]
```diff
@@ -46,18 +46,21 @@ To modify PATH, just follow the pictures after going to the ``Control Panel``:
...
 
 Then, go to ``Advanced`` > ``Environment Variables...``:
 
 .. image:: ./_static/images/screenshot-advanced-system-settings.png
    :align: center
    :target: ./_static/images/screenshot-advanced-system-settings.png
+   :alt: A screenshot of the System Properties window
```
Comment (StrikerRUS) [inline docs/GPU-Windows.rst:56]:
```suggestion
   :alt: A screenshot of the System Properties window.
```

File: docs/GPU-Windows.rst [hunk 11]
```diff
@@ -46,18 +46,21 @@ To modify PATH, just follow the pictures after going to the ``Control Panel``:
...
 
 Under ``System variables``, the variable ``Path``:
 
 .. image:: ./_static/images/screenshot-environment-variables.png
    :align: center
    :target: ./_static/images/screenshot-environment-variables.png
+   :alt: A screenshot of the Environment variables window with variable path selected under the system variables
```
Comment (StrikerRUS) [inline docs/GPU-Windows.rst:63]:
```suggestion
   :alt: A screenshot of the Environment variables window with variable path selected under the system variables.
```

File: docs/GPU-Windows.rst [hunk 12]
```diff
@@ -105,6 +108,7 @@ You may choose a version other than the most recent one if you need a previous M
 .. image:: ./_static/images/screenshot-mingw-installation.png
    :align: center
    :target: ./_static/images/screenshot-mingw-installation.png
+   :alt: A screenshot of the Min G W installation setup settings window
```
Comment (StrikerRUS) [inline docs/GPU-Windows.rst:111]:
```suggestion
   :alt: A screenshot of the Min G W installation setup settings window.
```

File: docs/GPU-Windows.rst [hunk 13]
```diff
@@ -123,6 +127,7 @@ You can check which MinGW version you are using by running the following in a co
 .. image:: ./_static/images/screenshot-r-mingw-used.png
    :align: center
    :target: ./_static/images/screenshot-r-mingw-used.png
+   :alt: A screenshot of the administrator command prompt where G C C version is being checked
```
Comment (StrikerRUS) [inline docs/GPU-Windows.rst:130]:
```suggestion
   :alt: A screenshot of the administrator command prompt where G C C version is being checked.
```

File: docs/GPU-Windows.rst [hunk 14]
```diff
@@ -220,6 +225,7 @@ This is what you should (approximately) get at the end of Boost compilation:
 .. image:: ./_static/images/screenshot-boost-compiled.png
    :align: center
    :target: ./_static/images/screenshot-boost-compiled.png
+   :alt: A screenshot of the command prompt that ends with text that reads - updated 14621 targets
```
Comment (StrikerRUS) [inline docs/GPU-Windows.rst:228]:
```suggestion
   :alt: A screenshot of the command prompt that ends with text that reads - updated 14621 targets.
```

File: docs/GPU-Windows.rst [hunk 15]
```diff
@@ -243,6 +249,7 @@ Installing Git for Windows is straightforward, use the following `link`_.
 .. image:: ./_static/images/screenshot-git-for-windows.png
    :align: center
    :target: ./_static/images/screenshot-git-for-windows.png
+   :alt: A screenshot of the website to download git that shows various versions of git compatible with 32 bit and 64 bit Windows separately
```
Comment (StrikerRUS) [inline docs/GPU-Windows.rst:252]:
```suggestion
   :alt: A screenshot of the website to download git that shows various versions of git compatible with 32 bit and 64 bit Windows separately.
```

File: docs/GPU-Windows.rst [hunk 16]
```diff
@@ -270,6 +277,7 @@ Installing CMake requires one download first and then a lot of configuration for
 .. image:: ./_static/images/screenshot-downloading-cmake.png
    :align: center
    :target: ./_static/images/screenshot-downloading-cmake.png
+   :alt: A screenshot of the binary distributions of C Make for downloading on 64 bit Windows
```
Comment (StrikerRUS) [inline docs/GPU-Windows.rst:280]:
```suggestion
   :alt: A screenshot of the binary distributions of C Make for downloading on 64 bit Windows.
```

File: docs/GPU-Windows.rst [hunk 17]
```diff
@@ -288,16 +296,19 @@ Installing CMake requires one download first and then a lot of configuration for
    .. image:: ./_static/images/screenshot-create-directory.png
       :align: center
       :target: ./_static/images/screenshot-create-directory.png
+      :alt: A screenshot with a pop-up window that reads - Build directory does not exist, should I recreate it?
```
Comment (StrikerRUS) [inline docs/GPU-Windows.rst:299]:
```suggestion
      :alt: A screenshot with a pop-up window that reads - Build directory does not exist, should I create it?
```

File: docs/GPU-Windows.rst [hunk 18]
```diff
@@ -288,16 +296,19 @@ Installing CMake requires one download first and then a lot of configuration for
    .. image:: ./_static/images/screenshot-create-directory.png
       :align: center
       :target: ./_static/images/screenshot-create-directory.png
+      :alt: A screenshot with a pop-up window that reads - Build directory does not exist, should I recreate it? 
 
    .. image:: ./_static/images/screenshot-mingw-makefiles-to-use.png
       :align: center
       :target: ./_static/images/screenshot-mingw-makefiles-to-use.png
+      :alt: A screenshot that asks to sepcify the generator for the project which should be selected as Min G W makefiles and selected as the use default native compilers option
```
Comment (StrikerRUS) [inline docs/GPU-Windows.rst:304]:
```suggestion
      :alt: A screenshot that asks to specify the generator for the project which should be selected as Min G W makefiles and selected as the use default native compilers option.
```

File: docs/GPU-Windows.rst [hunk 19]
```diff
@@ -306,6 +317,7 @@ Installing CMake requires one download first and then a lot of configuration for
    .. image:: ./_static/images/screenshot-configured-lightgbm.png
       :align: center
       :target: ./_static/images/screenshot-configured-lightgbm.png
+      :alt: A screenshot of the C Make window after clicking on the configure button
```
Comment (StrikerRUS) [inline docs/GPU-Windows.rst:320]:
```suggestion
      :alt: A screenshot of the C Make window after clicking on the configure button.
```

File: docs/GPU-Windows.rst [hunk 20]
```diff
@@ -366,6 +378,7 @@ You can do everything in the Git Bash console you left open:
 .. image:: ./_static/images/screenshot-lightgbm-with-gpu-support-compiled.png
    :align: center
    :target: ./_static/images/screenshot-lightgbm-with-gpu-support-compiled.png
+      :alt: A screenshot of the git bash window with Light G B M successfully installed
```
Comment (StrikerRUS) [inline docs/GPU-Windows.rst:381]:
```suggestion
   :alt: A screenshot of the git bash window with Light G B M successfully installed.
```

File: docs/GPU-Windows.rst [hunk 21]
```diff
@@ -382,6 +395,7 @@ You can now test LightGBM directly in CLI in a **command prompt** (not Git Bash)
 .. image:: ./_static/images/screenshot-lightgbm-in-cli-with-gpu.png
    :align: center
    :target: ./_static/images/screenshot-lightgbm-in-cli-with-gpu.png
+      :alt: A screenshot of the command prompt where a binary classification model is being trained using Light G B M
```
Comment (StrikerRUS) [inline docs/GPU-Windows.rst:398]:
```suggestion
   :alt: A screenshot of the command prompt where a binary classification model is being trained using Light G B M.
```

File: docs/GPU-Windows.rst [hunk 22]
```diff
@@ -397,6 +411,7 @@ Now that you compiled LightGBM, you try it... and you always see a segmentation
 .. image:: ./_static/images/screenshot-segmentation-fault.png
    :align: center
    :target: ./_static/images/screenshot-segmentation-fault.png
+   :alt: A screenshot of the command prompt where a segmentation fault has occurred while using Light G B M
```
Comment (StrikerRUS) [inline docs/GPU-Windows.rst:414]:
```suggestion
   :alt: A screenshot of the command prompt where a segmentation fault has occurred while using Light G B M.
```

File: docs/GPU-Windows.rst [hunk 23]
```diff
@@ -411,19 +426,22 @@ You will have to redo the compilation steps for LightGBM to add debugging mode.
 .. image:: ./_static/images/screenshot-files-to-remove.png
    :align: center
    :target: ./_static/images/screenshot-files-to-remove.png
+   :alt: A screenshot of the Light G B M folder with 1 folder and 3 files selected to be removed
```
Comment (StrikerRUS) [inline docs/GPU-Windows.rst:429]:
```suggestion
   :alt: A screenshot of the Light G B M folder with 1 folder and 3 files selected to be removed.
```

File: docs/GPU-Windows.rst [hunk 24]
```diff
@@ -411,19 +426,22 @@ You will have to redo the compilation steps for LightGBM to add debugging mode.
...
 
 Once you removed the file, go into CMake, and follow the usual steps.
 Before clicking "Generate", click on "Add Entry":
 
 .. image:: ./_static/images/screenshot-added-manual-entry-in-cmake.png
    :align: center
    :target: ./_static/images/screenshot-added-manual-entry-in-cmake.png
+   :alt: A screenshot of the Cache Entry popup where the name is set to C Make_Build_Type in all caps, the type is set to STRING in all caps and the value is set to Debug
```
Comment (StrikerRUS) [inline docs/GPU-Windows.rst:437]:
```suggestion
   :alt: A screenshot of the Cache Entry popup where the name is set to CMAKE_BUILD_TYPE in all caps, the type is set to STRING in all caps and the value is set to Debug.
```

File: docs/GPU-Windows.rst [hunk 25]
```diff
@@ -411,19 +426,22 @@ You will have to redo the compilation steps for LightGBM to add debugging mode.
...
 
 In addition, click on Configure and Generate:
 
 .. image:: ./_static/images/screenshot-configured-and-generated-cmake.png
    :align: center
    :target: ./_static/images/screenshot-configured-and-generated-cmake.png
+   :alt: A screenshot of the C Make window after clicking on configure and generate
```
Comment (StrikerRUS) [inline docs/GPU-Windows.rst:444]:
```suggestion
   :alt: A screenshot of the C Make window after clicking on configure and generate.
```

File: docs/GPU-Windows.rst [hunk 26]
```diff
@@ -437,6 +455,7 @@ open a command prompt and run the following:
 .. image:: ./_static/images/screenshot-debug-run.png
    :align: center
    :target: ./_static/images/screenshot-debug-run.png
+   :alt: A screenshot of the command prompt after the command above is run
```
Comment (StrikerRUS) [inline docs/GPU-Windows.rst:458]:
```suggestion
   :alt: A screenshot of the command prompt after the command above is run.
```

**Review by StrikerRUS — CHANGES_REQUESTED** [review]
Very nice and detailed descriptions!
Please check some typos and add dots at the end for the consistency with descriptions in other files.

**Comment by akshitadixit** [comment]
> Very nice and detailed descriptions!
> Please check some typos and add dots at the end for the consistency with descriptions in other files.

Done the needful.

**Comment by jameslamb** [comment]
Looks good to me! @StrikerRUS I didn't want to overstep and dismiss your blocking review, so you'll have to approve before this can be merged.

**Review by StrikerRUS — APPROVED** [review]
@akshitadixit Thanks a lot for the contribution that required a thoughtful research and for quick fixes!

**Pull request merged**